### PR TITLE
103 del btn menu

### PR DESCRIPTION
**Pull request by yishain11**

### CODE REVIEW

File: frontend/src/components/AgreementCard.tsx [hunk 1]
```diff
@@ -84,6 +86,28 @@ const AgreementCard: FC<IAgreementCardProps> = ({
...
+        action: () => archiveAgreement(id, !isArchived),
+      },
+    ];
+    if (role && role === 'moderator') {
+      menuItems.push({
+        text: 'Delete',
+        icon: <TrashEL />,
+        action: () => {
+          handleClickOpenDialog();
+        },
```
Comment (avirankatz) [inline frontend/src/components/AgreementCard.tsx:104]: This could be a one-liner

File: frontend/src/components/AgreementCard.tsx [hunk 2]
```diff
@@ -113,21 +137,7 @@ const AgreementCard: FC<IAgreementCardProps> = ({
...
-                    {
-                      text: 'Delete',
-                      icon: <TrashEL />,
-                      action: () => {
-                        handleClickOpenDialog();
-                      },
-                      textColor: '#FC6D8F',
-                    },
-                  ]}
+                  menuItems={getMenuItems()}
```
Comment (avirankatz) [inline frontend/src/components/AgreementCard.tsx:140]: Great refactor

**Review by avirankatz — CHANGES_REQUESTED** [review]
Great PR. Please see the requested change

File: frontend/src/components/AgreementCard.tsx (hunk 2 above)
Comment (yishain11) [reply frontend/src/components/AgreementCard.tsx:140]: thx

**Review by avirankatz — APPROVED** [review]

**Pull request merged**